src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/attn_memcpy.cpp

```diff
@@ -83,13 +83,16 @@ static void paged_attn_memcpy_kernel(const ov::intel_cpu::PlainTensor& k_input,
                                      const ov::intel_cpu::PlainTensor& past_v_output,
                                      const ov::intel_cpu::PlainTensor& slot_mapping) {
     size_t B = k_input.m_dims[0], H = k_input.m_dims[1], L1 = k_input.m_dims[2], S = k_input.m_dims[3];
+    size_t block_size = past_k_output.m_dims[2];
     parallel_for3d(B, L1, H, [&](size_t b, size_t m, size_t h) {
-        auto block_idx = slot_mapping.ptr<int32_t>(b)[m];
-        if (block_idx < 0) return;
-        attn_copy(past_k_output.ptr<T2>(block_idx, h, 0),
+        auto slot = slot_mapping.ptr<int32_t>(b)[m];
+        if (slot < 0) return;
+        auto block_number = slot / block_size;
+        auto block_offset = slot % block_size;
+        attn_copy(past_k_output.ptr<T2>(block_number, h, block_offset, 0),
                   k_input.ptr<T>(b, h, m, 0),
                   S);
-        attn_copy(past_v_output.ptr<T2>(block_idx, h, 0),
+        attn_copy(past_v_output.ptr<T2>(block_number, h, block_offset, 0),
```

Comment:
Just for the record, as Pointed by @zhangYiIntel , these parallel memory copy put current kv-token in dirty caches of different processor(physical cores), and later on parallel computation of MHA would access them on yet another  processor which induce great penalty (similar to one-time false sharing)

Reply:
It should not. The copied memory here may also be accessed by different cores in the next loop.